### PR TITLE
FTUE TypeScript

### LINKED ISSUE #413: TSify the frontend
**Is your feature request related to a problem? Please describe.**
We should move to TS before launch to help our code stability (ex: can catch property-related issues at ts-compile time vs runtime)

**Additional context**
Check Send's approach to TS and emulate closely so we make the same architectural choices between the two projects to reduce cognitive load.

### LINKED ISSUE #625: Sync Calendars stuck on loading
**Describe the bug**
When clicking on `Sync Calendars` in the Calendar Settings page, the button keep spinning and nothing happens.
In the JS console I get this error as soon as I click on the sync button:
```
[Vue warn]: Error in https://vuejs.org/error-reference/#runtime-6: "TypeError: i.calendars.value is undefined"
```

**To Reproduce**

1. Go to `Settings` > `Calendar`
2. Click on `Sync Calendars`

**Expected behavior**
The sync should be pretty fast and available calendars should be listed.

**Actual behavior**
The button is stuck spinning.

**Screenshots**
image

**System:**
- OS: Windows 11
- Browser Version: Firefox 129.0.2 64-bit

### PR DESCRIPTION
**Pull request by devmount**
## Description of the Change

This change adds types for all FTUE components.

- The script section was moved to the top in the vue files
- Type errors from regressions got fixed
- Type errors about mandatory/optional component properties got fixed
- All deprecated definitions were moved or deleted

## Applicable Issues

Closes #413
Closes #625

### CODE REVIEW

File: frontend/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "emitDecoratorMetadata": true,
     "experimentalDecorators": true,
     "allowSyntheticDefaultImports": true,
-    "target": "es2017",
+    "target": "es2021",
```
Comment (devmount) [inline frontend/tsconfig.json:10]: This was necessary because we're using String.prototype.replaceAll().

**Review by aaspinwall — COMMENTED** [review]
Looking good, just a comment to improve readability

File: frontend/src/components/FTUE/SetupSchedule.vue [hunk 2]
```diff
@@ -80,22 +33,20 @@ const { connectedCalendars } = storeToRefs(calendarStore);
 const { schedules } = storeToRefs(scheduleStore);
 const { timeToBackendTime, timeToFrontendTime } = scheduleStore;
 
-const calendarOptions = computed(() => connectedCalendars.value.map((calendar) => ({
+const calendarOptions = computed<SelectOption[]>(() => connectedCalendars.value.map((calendar) => ({
   label: calendar.title,
   value: calendar.id,
 })));
-const durationOptions = [15, 30, 45, 60, 75, 90].map((min) => ({
+const durationOptions: SelectOption[] = [15, 30, 45, 60, 75, 90].map((min) => ({
```
[the remark this answers is not in the document]
Comment (devmount) [reply frontend/src/components/FTUE/SetupSchedule.vue]: Great suggestion. I added a corresponding constant to our definitions.ts and also refactored existing constants to upper snake case.

**Review by aaspinwall — APPROVED** [review]
looking great!

**Review by MelissaAutumn — APPROVED** [review]

**Pull request merged**